src/crate/client/doctests/sqlalchemy.txt

```diff
@@ -0,0 +1,379 @@
+
+But it is also possible to just select a part of the document, even inside the
+``Object`` type::
+
+    >>> sorted(session.query(Character.details['gender']).all())
+    [('female',), ('male',)]
+
+In addition, filtering on the attributes inside the ``details`` column is also
+possible::
+
+    >>> query = session.query(Character.name)
+    >>> query.filter(Character.details['gender'] == 'male').all()
+    [('Arthur Dent',)]
+
+Updating Complex Types
+----------------------
+
+The SQLAlchemy CrateDB dialect supports change tracking deep down the nested
+levels of a ``Object`` type field. For example the following query will only
+update the ``gender`` key. The ``species`` key which is on the same level will
+be left untouched.
+
+::
+
+    >>> char = session.query(Character).filter_by(name='Arthur Dent').one()
+    >>> char.details['gender'] = 'manly man'
+    >>> session.commit()
+    >>> session.refresh(char)
+
+    >>> char.details['gender']
+    u'manly man'
+
+    >>> char.details['species']
+    u'human'
+
+Object Array
+------------
+
+In addition to the ``Object`` type the CrateDB sqlalchemy dialect also includes
+a type called ``ObjectArray``. This type maps to a Python list of dictionaries.
+
+Note that opposed to the ``Object`` type the ``ObjectArray`` type isn't smart
+and doesn't have an intelligent change tracking. Therefore the generated UPDATE
+statement will affect the whole list::
+
+    >>> char.more_details = [{'foo': 1, 'bar': 10}, {'foo': 2}]
+    >>> session.commit()
+
+    >>> char.more_details.append({'foo': 3})
+    >>> session.commit()
+
+This will generate an UPDATE statement roughly like this::
+
+    "UPDATE characters set more_details = ? ...", ([{'foo': 1, 'bar': 10}, {'foo': 2}, {'foo': 3}],)
+
+    >>> refresh("characters")
+
+To do queries against fields of ``ObjectArray``s you have to use the
+``.any(value, operator=operators.eq)`` method on a subscript, because accessing
+fields of object arrays (e.g. ``Character.more_details['foo']``) returns an
+array of the field type.
+
+Only one of the objects inside the array has to match in order for the result
+to be returned::
+
+    >>> from sqlalchemy.sql import operators
+    >>> query = session.query(Character.name)
+    >>> query.filter(Character.more_details['foo'].any(1, operator=operators.eq)).all()
+    [(u'Arthur Dent',)]
+
+Querying a field of an object array will result in an array of
+all values of that field of all objects in that object array::
+
+    >>> query = session.query(Character.more_details['foo']).order_by(Character.name)
+    >>> query.all()
+    [([1, 2, 3],), (None,)]
+
+Count and Group By
+==================
+
+SQLAlchemy supports different approaches to issue a query with a count
+aggregate function. Take a look at the `Counting section in the tutorial
+<http://docs.sqlalchemy.org/en/latest/orm/tutorial.html#counting>`_ for a full
+overview.
+
+CrateDB currently doesn't support all variants as it can't handle the
+sub-queries yet.
```

Comment:
@chaudum this isn't true with recent versions of CrateDB, so does this need changing?

Reply:
You're right, this limitation is gone now.